src/Microsoft.AspNetCore.Server.Kestrel/Http/ListenerPrimary.cs

```diff
@@ -101,12 +112,59 @@ protected override void DispatchConnection(UvStreamHandle socket)
             }
         }
 
+        private void DetachFromIOCP(UvHandle handle)
+        {
+            if (!_tryDetachFromIOCP)
+            {
+                return;
+            }
+
+            // https://msdn.microsoft.com/en-us/library/windows/hardware/ff728840(v=vs.85).aspx
+            const int FileReplaceCompletionInformation = 61;
+            // https://msdn.microsoft.com/en-us/library/cc704588.aspx
+            const uint STATUS_INVALID_INFO_CLASS = 0xC0000003;
+
+            var statusBlock = new IO_STATUS_BLOCK();
+            var socket = IntPtr.Zero;
+            Thread.Loop.Libuv.uv_fileno(handle, ref socket);
+
+            if (NtSetInformationFile(socket, out statusBlock, _fileCompletionInfoPtr,
+                (uint)Marshal.SizeOf<FILE_COMPLETION_INFORMATION>(), FileReplaceCompletionInformation) == STATUS_INVALID_INFO_CLASS)
+            {
+                // Replacing IOCP information is only supported on Windows 8.1 or newer
+                _tryDetachFromIOCP = false;
+            }
+        }
+
+        private struct IO_STATUS_BLOCK
+        {
+            uint status;
+            ulong information;
+        }
+
+        private struct FILE_COMPLETION_INFORMATION
+        {
+            public IntPtr Port;
+            public IntPtr Key;
+        }
+
+        [DllImport("NtDll.dll")]
```

Comment:
Yes. I will however try to find api-set needed. It is very likely that this api set is not included in dotnet cli nor is installed on win7 by default. If this is the case we could ask cli folks to add it and we would not introduce another dependency on reverse forwarders.